### ISSUE TITLE
How to ZIP the Electron Mac files like Electron does it?

### ISSUE THREAD

**Issue by johnjohn1231** [electron/electron#11622]
When I use 7-Zip on Windows to extract the Mac Electron ZIP archive, add my files and then ZIP it back up, it loses some symlinks or something. There's some kind of difference between Windows and Mac that makes the resulting ZIP file not quite right. Specifically, if you download the following ZIP file straight to your Mac and extract it, the resulting program WILL run fine: https://github.com/electron/electron/releases/download/v1.7.10/electron-v1.7.10-darwin-x64.zip

BUT if you download the same file to Windows, extract it with 7-Zip, then ZIP it back up and send it to the Mac, the resulting program will FAIL to run.

With the Linux version, I had a similar issue, which was solved by making Linux's package a .tar instead of .zip. Same "trick" didn't work on Mac, however.

I've asked many people and searched endlessly without even finding this mentioned anywhere. I cannot understand why this happens or how to solve it. (And yes, I must use the Windows machine and that alone.)

I've also vacuumed the 7-Zip manual without finding any mention of this. Any ideas?

Basically, how are the Electron developers making the working Mac ZIP archive? Do they also use 7-Zip? Something else?

**Comment by felixrieseberg** [comment]
I heavily recommend building things for a platform using the target platform. That said, the whole thing is open source, you can find the whole release and build script corpus in the `script` folder. To see how zip files are created on macOS, check out https://github.com/electron/electron/blob/master/script/lib/util.py#L125.

**Comment by johnjohn1231** [comment]
I don't have a Mac, but even if I did, that would be highly impractical for obvious reasons?

**Comment by johnjohn1231** [comment]
As for that code, I don't understand it at all.
